core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -1071,13 +1071,16 @@ func (fm *FluxMonitor) initialRoundState() flux_aggregator_wrapper.OracleRoundSt
 }
 
 func (fm *FluxMonitor) queueTransactionForTxm(tx pg.Queryer, runID int64, answer decimal.Decimal, roundID uint32, log *flux_aggregator_wrapper.FluxAggregatorNewRound) error {
+	// Use pipeline run ID to generate globally unique key that can correlate this run to a Tx
+	idempotencyKey := fmt.Sprintf("fluxmonitor-%d", runID)
```

Comment:
Can you confirm with someone if runID here is unique across FluxMonitor?
I see there's also a roundID. 
The Pipeline Run has more ids in there: https://github.com/smartcontractkit/chainlink/blob/ed62e474aa50b0855be9dc856c7d0367b8062ca9/core/services/pipeline/models.go#L37

Maybe @samsondav can help confirm that.

Reply:
That's my understanding as well. I build the IdempotencyKey with the same runID inserted into `flux_monitor_round_stats_v2.pipeline_run_id`, which has a reference to the `pipeline_runs.id` column.
https://github.com/smartcontractkit/chainlink/blob/fe3f7d4705f670c88b0a7ef6db8df6f0f6f3a2f4/core/store/migrate/migrations/0013_create_flux_monitor_round_stats_v2.sql#L2-L10